apps/comps/app/api/sandbox/agents/route.ts

```diff
@@ -1,64 +1,34 @@
 import { NextRequest } from "next/server";
 
-import {
-  extractSessionCookie,
-  mainApiRequest,
-  sandboxAdminRequest,
-} from "@/app/api/sandbox/_lib/sandbox-config";
+import { sandboxAdminRequest } from "@/app/api/sandbox/_lib/sandbox-config";
 import {
   createSuccessResponse,
   withErrorHandling,
 } from "@/app/api/sandbox/_lib/sandbox-response";
-import {
-  AdminCreateAgentResponse,
-  CreateAgentRequest,
-  ProfileResponse,
-} from "@/types";
+import { AdminCreateAgentRequest, AdminCreateAgentResponse } from "@/types";
 
 /**
  * POST /api/sandbox/agents
- * Creates an agent in the sandbox environment by:
- * 1. Fetching user profile from the base API using session cookie to get walletAddress
- * 2. Creating agent in sandbox using the admin API
+ * Creates an agent in the sandbox environment using the provided user and agent data
  */
 async function handleCreateAgent(request: NextRequest) {
```

Comment:
we save a call to the "main" prod API by using the user session context, which already has user profile info